src/lib/libdylink.js

```diff
@@ -1159,9 +1159,9 @@ var LibraryDylink = {
 #if ASYNCIFY
   _dlopen_js__async: true,
 #endif
-  _dlopen_js: (handle) => {
+  _dlopen_js: {{{ asyncIf(ASYNCIFY == 2) }}} (handle) => {
```

Comment:
Does `__async` not imply this?  If not, what *does* `__async` do?

Reply:
`__async` just tells emscripten to wrap the import with JSPI. Not requiring `async function` was an optimization that allowed for synchronous returns. However, JSPI is now going to suspend no matter what the return value is, so we could now probably change it to imply `async` or require the function to be marked `async`.

We talked about getting rid of `__async`, and just using `async function`, but there are cases where we don't actually want to wrap with JSPI if the function is `async`. More discussion in https://github.com/emscripten-core/emscripten/issues/20289

Reply:
I'd say land this first. I probably won't have time to look into the bigger overhaul for a bit.

Reply:
Fair enough.  Resolve this then